functions/go/apply-replacements/replacements.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"io"
+
+	sdk "github.com/GoogleContainerTools/kpt-functions-catalog/thirdparty/kyaml/fnsdk"
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+)
+
+type Replacements struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+const fnConfigKind = "Replacements"
+
+// Config initializes Replacements from a functionConfig sdk.KubeObject
+func (r *Replacements) Config(functionConfig *sdk.KubeObject) error {
+	if functionConfig.Kind() != fnConfigKind {
+		return fmt.Errorf("received functionConfig of kind %s, only functionConfig of kind %s is supported",
+			functionConfig.Kind(), fnConfigKind)
+	}
+	r.Replacements = []types.Replacement{}
+	if err := functionConfig.As(r); err != nil {
+		return fmt.Errorf("unable to convert functionConfig to %s:\n%w",
+			"replacements", err)
+	}
+	return nil
+}
+
+// Run accepts a reader and writer, and calls framework.Execute to run the function. This exists
+// to enable the function to be invoked as a library.
+func (r *Replacements) Run(reader io.Reader, writer io.Writer) error {
+	rw := &kio.ByteReadWriter{
+		Reader:                reader,
+		Writer:                writer,
+		KeepReaderAnnotations: true,
```

Comment:
Got it. I added this here because I thought (maybe incorrectly) that in standup we said that we wanted to have a `Run` method similar to the [pkg_context builtin function](https://github.com/GoogleContainerTools/kpt/blob/43c9d295300018748339802542ad6704d10832db/internal/builtins/pkg_context.go#L51), to make the logic importable. 

If we want to have the `Run` method that takes in a reader and a writer, but we don't want to use `framework.Execute`, maybe we can add a `framework.Execute` to the fn sdk? However, `Run` is not strictly necessary, so I have removed it, and I made `ApplyReplacements(*fn.ResourceList)` in main.go public. Kpt/porch will be able to call this function if they would like to import the replacements logic. 

@yuwenma @droot FYI.

Reply:
Further discussion is in https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/768#discussion_r826278649